spec/support/matchers/be_a_response_like.rb

```diff
@@ -42,13 +52,39 @@
 
   diffable
 
+  summary = []
   failure_message do |actual|
-    bad_keys_info = (!!@problem_keys ? "Bad keys: #{@problem_keys}" : '')
+    begin
+      diffs = HashDiff.best_diff(expected, actual)
+      if diffs
+        diffs.each do |comparator, key, expected_value, actual_value|
+          case comparator
+          when '-'
+            summary << "- #{key}: #{truncate(expected_value, 80)}"
+          when '+'
+            summary << "+ #{key}: #{truncate(expected_value, 80)}"
+          when '~'
+            next if expected_value.is_a?(Regexp) && expected_value.match(actual_value) rescue false
+            summary << "! #{key}:"
+            if expected_value.is_a?(Regexp)
+              expected_value = expected_value.inspect
+            end
+            summary << "  - #{truncate(expected_value, 80)}"
+            summary << "  + #{truncate(actual_value, 80)}"
+          end
+        end
+      end
+    rescue => ex
+      $stderr.puts("Error in hashdiff: #{ex} \n #{ex.backtrace[0..5]}")
```

Comment:
@ljfranklin I thought given that the exception should be unlikely it would be better to write it directly to stderr, but given that in testing stderr can be redirected, good point.